mp/src/game/shared/momentum/run/mom_replay_factory.cpp

```diff
@@ -98,4 +120,159 @@ CMomReplayBase* CMomReplayFactory::LoadReplayFile(const char* pFileName, bool bF
     return toReturn;
 }
 
+#ifdef CLIENT_DLL
+LocalTimesStatus_t CMomReplayFactory::AsyncLoadReplayFileMultiple(const CUtlStringList& filenames, const char *pathID,
+                                                                 MomLocalReplayCallback_t callback,
+                                                                 MomReplayAsyncHandle* outHandle)
+{
+    auto *internalRequest = new MomReplayLoadRequestInternal;
+    if (outHandle)
+        *outHandle = internalRequest->handle;
+    requests.AddToTail(internalRequest);
+    internalRequest->callback = callback;
+
+    size_t count = filenames.Size();
+
+    auto fsControls = new FSAsyncControl_t[count];
+    auto fsRequests = new FileAsyncRequest_t[count]; // they SHOULD get deleted by filesystem
+
+    internalRequest->arrControls = fsControls;
+    internalRequest->controls_count = count;
+    internalRequest->controls_pending = count;
+
+    FOR_EACH_VEC (filenames, i)
+    {
+        fsRequests[i].pszFilename = filenames[i];
+        fsRequests[i].pfnCallback = FilesystemCallbackForward;
+        fsRequests[i].pszPathID = pathID;
+        fsRequests[i].pContext = internalRequest->handle;
+    }
+
+    if (filesystem->AsyncReadMultiple(fsRequests, count, fsControls) != FSASYNC_OK)
+    {
+        Warning("Bad async replay load request!");
+        AsyncCleanup(internalRequest->handle);
+        return LocalTimesStatus_t::LOCALREPLAY_BADREQUEST;
+    }
+
+    return LocalTimesStatus_t::LOCALREPLAY_INPROGRESS;
+
+}
+void CMomReplayFactory::FilesystemCallbackForward(const FileAsyncRequest_t &request, int nBytesRead,
+                                                  FSAsyncStatus_t err)
+{
+    g_ReplayFactory.FilesystemCallback(request, nBytesRead, err);
+}
+void CMomReplayFactory::FilesystemCallback(const FileAsyncRequest_t &request, int nBytesRead, FSAsyncStatus_t err)
```

Comment:
`err` is kinda misleading as a variable name, prefer `status`:
```suggestion
void CMomReplayFactory::FilesystemCallback(const FileAsyncRequest_t &request, int nBytesRead, FSAsyncStatus_t status)
```

Reply:
It's basically copy-paste from the typedef from filesystem.h but I agree